### PR TITLE
fix: clipboard button ui interaction

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://github.com/devtron-labs/devtron/issues/4881

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ClusterNodes/NodeDetails.tsx [hunk 1]
```diff
@@ -255,29 +254,9 @@ export default function NodeDetails({
...
-                            onClick={() => {
-                                copyToClipboard(`${key}=${value || ''}`, () => {
-                                    setCopied(true)
-                                })
-                            }}
-                        />
-                    </div>
-                </Tippy>
+                <div className="dc__visible-hover--child">
+                    <ClipboardButton content={`${key}=${value || ''}`} copiedTippyText={'Copied!'} duration={1000} />
```
Comment (AbhishekA1509) [inline src/components/ClusterNodes/NodeDetails.tsx:258]: Encapsulate the string: ={`${key}=${value || ''}`}  into const

File: src/components/ciPipeline/ConfigureWebhook.tsx [hunk 2]
```diff
@@ -60,32 +44,44 @@ export const ConfigureWebhook = ({
...
-                                onClick={() => {
-                                    copyToClipboard(gitHost.webhookUrl, () => setCopiedUrl(true))
-                                }}
-                            >
-                                Click to copy webhook URL
-                                <Copy className="icon-dim-16 ml-4" />
+                        <div
+                            className="bcn-0 pt-6 pb-6 pl-12 pr-12 pt-6 pb-2 br-4 bw-1 en-2 mr-12 flex left cursor"
+                            data-testid="build-copy-webhook-url-button"
+                            onClick={() => {
```
Comment (AbhishekA1509) [inline src/components/ciPipeline/ConfigureWebhook.tsx:50]: Create a separate handler method for this.

File: src/components/app/details/cicdHistory/History.components.tsx [hunk 3]
```diff
@@ -168,14 +167,16 @@ export const GitChanges = ({
                     >
                         <div className="flex column left hover-trigger">
                             <div className="cn-9 fs-14 flex left">
-                                <CopyTippyWithText
-                                    copyText={extractImage(artifact)}
-                                    copied={copied}
-                                    setCopied={setCopied}
-                                />
+                                {extractImage(artifact)}
```
Comment (AbhishekA1509) [inline src/components/app/details/cicdHistory/History.components.tsx:170]: can we create a const variable calling this function and use them

File: src/components/ciPipeline/Webhook/WebhookDetailsModal.tsx [hunk 4]
```diff
@@ -517,33 +480,16 @@ export const WebhookDetailsModal = ({ close }: WebhookDetailType) => {
...
-                                style={{ right: '8px' }}
-                                onClick={() => {
-                                    copyToClipboard(value, () => {
-                                        setCopied(true)
-                                    })
-                                }}
-                            />
-                        </div>
-                    </Tippy>
+                    <div>
```
Comment (AbhishekA1509) [inline src/components/ciPipeline/Webhook/WebhookDetailsModal.tsx:489]: Check if we can remove this div

**Review by AbhishekA1509 — APPROVED** [review]

File: src/components/common/FloatingVariablesSuggestions/SuggestionItem.tsx [hunk 5]
```diff
@@ -88,7 +88,7 @@ export default function SuggestionItem({
 
                     <ClipboardButton
                         content={`@{{${variableName}}}`}
-                        copiedTippyText={`Copied: @{{${variableName}}}`}
+                        copiedTippyText={'Copied!'}
```
Comment (AbhishekA1509) [inline src/components/common/FloatingVariablesSuggestions/SuggestionItem.tsx:91]: Use "Copied !" instead {'Copied!'}

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 6]
```diff
@@ -218,17 +204,15 @@ const NodeComponent = ({
             return (
                 <>
                     {portList.map((val, idx) => {
+                        const copyText = `${node.name}.${node.namespace}:${val}`
                         if (idx > 0) {
                             return (
                                 <div className="flex left cn-9 m-0 dc__no-decore">
                                     <div className="" key={node.name}>
                                         {node.name}.{node.namespace}:{val}
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:212]: we should use copyText here

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 7]
```diff
@@ -239,20 +223,16 @@ const NodeComponent = ({
         }
         const portNumberPlaceHolder = (node) => {
             if (node.port?.length > 1) {
+                const copyText = `${node.name}.${node.namespace}:${node.port[0]}`
                 return (
                     <>
                         <div>
                             <span>
                                 {node.name}.{node.namespace}:{node.port[0]}
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:231]: Same here

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 8]
```diff
@@ -282,30 +262,7 @@ const NodeComponent = ({
...
-                <span>
-                    <Clipboard
-                        className="resource-action-tabs__clipboard icon-dim-12 pointer ml-8 mr-8"
-                        onClick={(event) => {
-                            toggleClipBoard(event, nodeName.split(' ').join(''))
-                        }}
-                    />
-                </span>
-            )
+            return <ClipboardButton content={nodeName} />
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:265]: is it not required? `nodeName.split(' ').join('')`

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**